### PR TITLE
New version

### PR DESCRIPTION
**Pull request by benijo**
## New version

- lots of bc breaks (added typehints to interface)
- support only `2.5.x` tracy

### TracyLoggingExtension

- keep original Tracy Logger in DIC with different key `logging.originalLogger`

### SendMailLogger

- not filter only `Exception` [#22]
- sending also `string` messages with `critical`/`error`/`exception` priority (as in the original Logger)
- configurable priority

### SentryLogger
- removed `final`
- added priority level
- sending also `string` messages with `critical`/`error`/`exception` priority
- configurable priority

### FileLogger

- renamed from `ExceptionFileLogger` (because it does not log only exceptions but`<priority>.log`)

### NullLogger

- with Tracy/ILogger interface [#10]

### Utils 

- used static methods `formatLogLine`,`formatMessage` from original `Tracy\Logger`

------------------

### TODO 

- [ ] tag new version (bc breaks)

### CODE REVIEW

**Comment by f3l1x** [comment]
Keep going. 👍

**Comment by f3l1x** [comment]
Any updates?

File: src/SendMailLogger.php [hunk 1]
```diff
@@ -32,19 +33,27 @@ public function setMailer(IMailer $mailer): void
 
 	/**
 	 * @param mixed $message
-	 * @param string $priority
 	 */
-	public function log($message, $priority): void
+	public function log($message, string $priority = ILogger::INFO): void
 	{
 		if (!in_array($priority, [ILogger::ERROR, ILogger::EXCEPTION, ILogger::CRITICAL], true)) return;
```
Comment (f3l1x) [inline src/SendMailLogger.php]: Idea: Maybe make it configurable?

File: src/Sentry/SentryLogger.php [hunk 2]
```diff
@@ -2,42 +2,74 @@
...
-	 * @param string|Exception $message
-	 * @param string $priority
+	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint
+	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingReturnTypeHint
+	 * @param mixed $message
 	 */
-	public function log($message, $priority): void
+	public function log($message, string $priority = ILogger::INFO): void
 	{
 		if (!in_array($priority, [ILogger::ERROR, ILogger::EXCEPTION, ILogger::CRITICAL], true)) return;
```
Comment (f3l1x) [inline src/Sentry/SentryLogger.php]: Idea: Maybe make it configurable?

File: src/SendMailLogger.php (hunk 1 above)
Comment (benijo) [reply src/SendMailLogger.php]: good idea

File: src/Sentry/SentryLogger.php (hunk 2 above)
Comment (benijo) [reply src/Sentry/SentryLogger.php]: its done

**Pull request merged**